src/node/kernel_notifications.cpp

```diff
@@ -52,6 +52,7 @@ kernel::InterruptResult KernelNotifications::blockTip(SynchronizationState state
 {
     {
         LOCK(m_tip_block_mutex);
+        Assert(index.GetBlockHash() != uint256{});
         m_tip_block = index.GetBlockHash();
```

Comment:
I think this should either be an `assert` or an `Assume` instead.
> Assume
>  * Should be used to run non-fatal checks. In debug builds it behaves like
> *   Assert()/assert() to notify developers and testers about non-fatal errors.
>  *   In production it doesn't warn or log anything.

And in most places I saw the `!....IsNull()` pattern instead:
https://github.com/bitcoin/bitcoin/blob/master/src/txdb.cpp#L97, https://github.com/bitcoin/bitcoin/blob/master/src/wallet/scriptpubkeyman.cpp#L1000, https://github.com/bitcoin/bitcoin/blob/master/src/primitives/transaction.cpp#L77, https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp#L2490, etc have have

And it would be simpler to check after assignment.

```suggestion
        m_tip_block = index.GetBlockHash();
        assert(!m_tip_block.IsNull());
```

Reply:
I think newer code is trying to avoid `assert` to prevent opening production nodes open to attacks like https://bitcoincore.org/en/2024/10/08/disclose-blocktxn-crash/

Looking at `developer-notes.md` section on asserts, I think `Assume` is better. However it depends on whether a mistake here is dangerous, which hard to say.

I find `IsNull()` too ambiguous for use with an `std::optional`.

Reply:
> I find IsNull() too ambiguous for use with an std::optional.

My mistake, I meant to write `!m_tip_block->IsNull()`, i.e. it's not on the optional

Reply:
Or you only want to check that it's either empty or if not it's not 0? In that case this seems like the simplest indeed

Reply:
`GetBlockHash()` is not an `std::optional`, so I only want to make sure it's not `0`.

After this is checked, we assign the value to `m_tip_block` which _is_ an `std::optional`, but at that point I'm not worried that `m_tip_block` remains unset.